### ISSUE TITLE
Failing unit tests (75% of suite): CanCreateEmptyJSONFile, CanCreatesNewJSONReader, and CanReadJSONReader

### ISSUE THREAD

**Issue by dariusstrasel** [mechamecho/DataDrivenTesting#3]
Hi Nafissa,

I ran the unit tests now that the compile errors are resolved. Can you run these and see if you get similar results? We'll want to see if these bugs are due to the way the tests are written, or if its a class design issue.

CanCreateEmptyJSONFile:
```
Expected: No Exception to be thrown
  But was:  <System.NullReferenceException: Object reference not set to an instance of an object.
   at TestDataAccess.JSONFile.JSONFilePathValidation()
   at TestDataAccess.JSONFile..ctor()
   at TestDataAccess.Tests.JsonFile.CreateJSONFile(String filePath) in C:\Users\dstrasel\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\TestDataAccess.cs:line 69
   at TestDataAccess.Tests.JsonFile.<>c.<CanCreateEmptyJSONFile>b__3_0() in C:\Users\dstrasel\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\TestDataAccess.cs:line 20
   at NUnit.Framework.Constraints.ThrowsConstraint.VoidInvocationDescriptor.Invoke()
   at NUnit.Framework.Constraints.ThrowsConstraint.ExceptionInterceptor.Intercept(Object invocation)>
```

CanCreatesNewJSONReader
```
Expected: No Exception to be thrown
  But was:  <System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.
File name: 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at TestDataAccess.JSONReader..ctor(JSONFile jsonFile, String testDataKey, Int32 testDataIndex)
   at TestDataAccess.Tests.JsonFile.CreateJSONReader(JSONFile jsonFile, String testDataKey, Int32 testDataIndex) in C:\Users\dstrasel\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\TestDataAccess.cs:line 62
   at TestDataAccess.Tests.JsonFile.<>c__DisplayClass5_0.<CanCreatesNewJSONReader>b__0() in C:\Users\dstrasel\Documents\git\github\DataDrivenTesting\TestDataAccess.Tests\TestDataAccess.cs:line 44
   at NUnit.Framework.Constraints.ThrowsConstraint.VoidInvocationDescriptor.Invoke()
   at NUnit.Framework.Constraints.ThrowsConstraint.ExceptionInterceptor.Intercept(Object invocation)
```

CanReadJSONReader
```
System.IO.FileNotFoundException : Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.
```

**Comment by mechamecho** [comment]
These are the exceptions I get.

CanCreatesNewJSONReader()

`Test Name:	CanCreatesNewJSONReader()

Result Message:	
Expected: No Exception to be thrown
  But was:  <System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.
File name: 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at TestDataAccess.JSONReader..ctor(JSONFile jsonFile, String testDataKey, Int32 testDataIndex)
   at TestDataAccess.Tests.JsonFile.CreateJSONReader(JSONFile jsonFile, String testDataKey, Int32 testDataIndex) 
   at TestDataAccess.Tests.JsonFile.<>c__DisplayClass5_0.<CanCreatesNewJSONReader>b__0()testing\JsonReaderCsharp\DataDrivenTesting\TestDataAccess.Tests\
   at NUnit.Framework.Constraints.ThrowsConstraint.VoidInvocationDescriptor.Invoke()
   at NUnit.Framework.Constraints.ThrowsConstraint.ExceptionInterceptor.Intercept(Object invocation)

>

`
CanReadJSONReader()

`Test Name:	CanReadJSONReader()

Result Message:	System.IO.FileNotFoundException : Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.

`

These are the exceptions I get, I don't get the NullReferenceException.

**Comment by dariusstrasel** [comment]
I believe 36f5f3f886462b3cf1d071a9f4fa2596f453daf4 fixed the null reference error for CanCreateEmptyJSONFile().

Since we got the same errors for CanReadJSONReader and CanCreatesNewJSONReader, and they're both "Could not load file or assembly 'Newtonsoft.Json'": I believe the NuGet packages arent being loaded correctly into the Unit test solution. I'll try to reload the NuGet packages later today. I'd usually do this over CLI via:

```BASH
cd $solution-directory
rm -rf ./packages/
nuget restore
```

**Comment by mechamecho** [comment]
I see, I will try it out later as well!

**Comment by mechamecho** [comment]
When I try to use the 
`rm -rf ./packages/`
command on the command prompt I get this error : 

> 'rm' is not recognized as an internal or external command,
operable program or batch file.

**Comment by dariusstrasel** [comment]
No worries. It looks like the issue is that the Unit test solution is missing some dependencies in the .csproj. I will send a PR and update this issue.

**Comment by dariusstrasel** [comment]
These failures are fixed as of 36f5f3f886462b3cf1d071a9f4fa2596f453daf4, and 4a03655225b3abaef94eddb3dea08434201e4ef5
